ConsoleProject/src/view/Frontend.cs

```diff
@@ -185,22 +209,55 @@ public class Frontend {
                         Bitmap bmp2 = new Bitmap(image1);
 
                         List<Color> palette = new List<Color>();
+                        
+                        List<Color> completeColors = new List<Color>();
 
                         Console.WriteLine(bmp2.PixelFormat);
+                                                
+                        string logContent = "";
+                        
                         for (int y = 0; y < bmp2.Height; y++) {
                             for (int x = 0; x < bmp2.Height; x++) {
                                 Color c = bmp2.GetPixel(x, y);
                                 if (palette.IndexOf(c) == -1) {
                                     if (palette.Count < 256) palette.Add(c);
                                 }
+                                if (completeColors.IndexOf(c) == -1) completeColors.Add(c);
                             }
                         }
-
+                        
+                        logContent += "Saved " + palette.Count + " colors out of " + completeColors.Count + "\n";
+                        
+                        if (log) {
+                            Texture tex = images.Find(x => x.Name == Path.GetFileNameWithoutExtension(file));
+                            if (tex != null) {
+                                logContent += tex.Name + " original colors info:\n";
+                                for (int ci = 0; ci < tex.Colors.Count; ci++) {
+                                    Color c = tex.Colors[ci];
+                                    if (palette.IndexOf(c) == -1) {
+                                        logContent += "Color removed: " + c.ToString() + "\n";
+                                    } else {
+                                        logContent += "Color saved: " + c.ToString() + "\n";
+                                    }                                       
+                                }
+                            }
+                        }
+                        logContent += "\n\n";
+                        
                         while (palette.Count < 16) palette.Insert(0, Color.FromArgb(0, 0, 0, 0));
 
                         if (palette.Count > 16) while (palette.Count < 256) palette.Insert(0, Color.FromArgb(0, 0, 0, 0));
 
                         palette.Sort((x, y) => x.A.CompareTo(y.A));
+                        
+                        if (log) {
```

Comment:
We'll use a logging framework for this. Please add a TODO comment here: "replace with Logger"